rimage/depth_map_raw.go

```diff
@@ -42,50 +42,58 @@ func ParseRawDepthMap(fn string) (*DepthMap, error) {
 		}
 	}
 
-	return ReadRawDepthMap(bufio.NewReader(f))
+	return ReadDepthMap(bufio.NewReader(f))
 }
 
-// ReadRawDepthMap returns a depth map from the given reader.
-func ReadRawDepthMap(f *bufio.Reader) (*DepthMap, error) {
-	var err error
-	dm := DepthMap{}
-
-	rawWidth, err := _readNext(f)
+// ReadDepthMap returns a depth map from the given reader.
+func ReadDepthMap(r io.Reader) (*DepthMap, error) {
+	// We expect the first 8 bytes to be a magic number assigned to a depthmap type
+	firstBytes, err := _readNext(r)
 	if err != nil {
 		return nil, err
 	}
-	dm.width = int(rawWidth)
-
-	if rawWidth == 6363110499870197078 { // magic number for VERSIONX
-		return readDepthMapFormat2(f)
+	switch firstBytes {
+	case 6363110499870197078: // magic number for VERSIONX
```

Comment:
What is the difference between depth map VERSIONX, depth map viam, and depth map raw? Why do we have all of them?

Reply:
thank you!